### ISSUE TITLE
module_dirs left out salt-ssh, leaving custom ext_pillars and modules out of salt-ssh

### ISSUE THREAD

**Issue by whytewolf** [saltstack/salt#45535]
### Description of Issue/Question

The `module_dirs` is not copied from the master opts into salt-ssh opts which causes modules and pillars left out of the transfer. 

### Setup

The most straightforward test is a custom pillar in a module_dirs directory. 

I put this pillar file in /srv/salt/extmods/pillar/test.py
```
__virtualname__ = 'test_pillars'

def __virtual__():
  return __virtualname__

def ext_pillar( minion_id, pillar, *args, **kwargs ):

    my_pillar = {'external_pillar': 'test'}


    return my_pillar
```
and configured my master with these settings
```
ext_pillar:
  - test_pillars: []

module_dirs:
  - /srv/salt/extmods
```

### Steps to Reproduce Issue
see above setup and then run `salt-ssh <minion> pillar.items`

the result will be 
```
[DEBUG   ] Could not LazyLoad test_pillars.ext_pillar: 'test_pillars.ext_pillar' is not available.
[CRITICAL] Specified ext_pillar interface test_pillars is unavailable
```

### Versions Report
I started testing this in 2016.11.7. However, I have moved up to see if this was fixed in any version.

```
Salt Version:
           Salt: 2017.7.2

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  4 2017, 00:39:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.4.1708 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.5.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.4.1708 Core
```

### Possable fix?

In testing this I did find that if add these two lines into `salt/client/ssh/__init__.py` around line 913 it works, but am unsure if that is the direction that we want to pursue. 
```
            if 'module_dirs' in self.opts:
                opts_pkg['module_dirs'] = self.opts['module_dirs']
```

**Comment by whytewolf** [comment]
ZD-2151

**Comment by gtmanfred** [comment]
Does it just load the external module into the `salt` thin tarball?

I am assuming putting it in salt://_pillar also works?

It doesn't seem crazy to me to want to be able to load any module that the master has loaded for salt-ssh

@saltstack/team-core does anyone have an opinion about this?

Thanks,
Daniel

**Comment by whytewolf** [comment]
It currently does not load the modules_dir modules into the `salt` thin tarball. I missed that while I was looking at the code. 

Putting the pillar in `salt://_pillar` does work as that will sync the pillar into the `extension_modules` directory which is picked up currently.  

Thank you,
Thomas

**Comment by gtmanfred** [comment]
right, but with your addition, it just loads the file into the thin tarball?

**Comment by whytewolf** [comment]
unfortunately, I only tested with pillar originally. so even with my change modules do not get added to the `salt` thin tarball and modules are still broken.

**Comment by gtmanfred** [comment]
ok, yeah it would probably be useful to have those modules included in the thin tarball just probably want someone from @saltstack/team-ssh to comment and affirm that.

Thanks!
Daniel

**Comment by rickh563** [comment]
@gtmanfred, Who needs to be called from the team-ssh in order to take next-steps on this discussion?

**Comment by gtmanfred** [comment]
Any of them, i just want to make sure that it wasn't a deliberate choice to not include module_dirs, because what could end up happening is that modules meant specifically for the salt-master are copied over to the salt-ssh minion, which might not be intended.

**Comment by rickh563** [comment]
@gtmanfred ... I do not know who comprises the team.

### Request

Please directly call one of team members by name in order to advance the workflow on this ticket.

**Comment by gtmanfred** [comment]
@saltstack/team-ssh is @thatch45 @cachedout and @rallytime

**Comment by thatch45** [comment]
Yes, you are doing this correctly, those modules should be included in the thin tarball and the code change proposed looks like it would be a viable fix here.

**Comment by rallytime** [comment]
@gtmanfred I'll assign this to you to be included in the 2017.7.4 release.
